pyproject.toml

```diff
@@ -6,7 +6,10 @@ build-backend = "hatchling.build"
 name = "jupyter_ai_monorepo"
 dynamic = ["version", "description", "authors", "urls", "keywords"]
 requires-python = ">=3.8"
```

Comment:
In general adding `<3.13` is not the best idea.

If we updated the pin on `jupyter-ai` and assuming that there is nothing about `jupyter-ai` that makes it break on 3.13 (which I think is the case) we would just annoy more users and packagers because once wheels for dependencies (`tokenizers`, `arrow`) become available for 3.13 we would be the package blocking them from upgrade (artificially). Also, the pip-solver would just install an older version of `jupyter-ai` instead.

However, this comment is on top-level `pyproject.toml` which is for developers only and here it does not really matter. Is updating the ranges in the monorepo-level `pyproject.toml` only what you are suggesting?

Reply:
Done in https://github.com/jupyterlab/jupyter-ai/pull/1029/commits/b8b5bdf578561553de16b4d301037d57a0de627d.

Reply:
And now it's failing...

Also see https://packaging.python.org/en/latest/guides/dropping-older-python-versions/#specify-the-version-ranges-for-supported-python-distributions:

> Avoid adding upper bounds to the version ranges, e. g. ">= 3.8, < 3.10". Doing so can cause different errors and version conflicts. See the [discourse-discussion](https://discuss.python.org/t/requires-python-upper-limits/12663) for more information.

I will remove the `<3.13` part as it is universally a bad idea. By this I also mean that it is bad for developers because if they try to install the monorepo (`pip install -e .`) on newer Python version to test if it works with say 3.14 alpha they will need to manually modify that file.

Reply:
@krassowski Thanks for finding that link on the PyPA site! I'm comfortable with the version ceiling being removed given that PyPA recommends against version ceilings on the Python version. Can you change this to just `>=3.9`?

Reply:
I see this is done as of the latest commit, thanks.